rtCamp/privacy-sandbox-demos
# Implement Legacy GSI Demo

## Description 

This pull request adds a new "Legacy GSI Demo" to demonstrate the new feature from the extension PSAT, GSI Detector. The Legacy GSI Demo relies on Third-party cookies to keep the user session. With this demo, the developer will be able to use the GSI detector to debug the application. 

## Relevant Technical Choices

- This pull request adds a new environment variable to add a Google Client ID 

## Testing Instructions
1. Clone the repository and checkout `feature/adding-gsi-demo` branch containing these changes.
2. Ensure you have Node.js and npm installed.
3. Run `npm install` to install all dependencies.
4. Set up the `.env` file to define a Google `ClientID`, to use the legacy GSI you must have a account from 2022
5. Start the server using `npm start`.
6. Navigate to the homepage and click on the link to the "Legacy GSI".
7. Click on the Sign in Button
8. Select a google account 
9. If the login succeed you name will be displayed on the page


## Screenshots

**Unlogged user**
<img width="1279" alt="Screenshot 2024-01-16 at 20 14 55" src="https://github.com/rtCamp/privacy-sandbox-demos/assets/330792/122cf8fc-640e-4711-9a54-6079e16f9b78">

**Logged user**
<img width="1272" alt="Screenshot 2024-01-16 at 19 46 25" src="https://github.com/rtCamp/privacy-sandbox-demos/assets/330792/92d2d7ff-d270-46a8-88bb-6e25ec9cb316">


**Extension Detection**

![Screenshot 2024-01-17 at 13 11 26](https://github.com/rtCamp/privacy-sandbox-demos/assets/330792/5fafcf51-f0ce-4343-aaa5-1611f3458a66)
